go/flags/endtoend/mysqlctld.txt

```diff
@@ -103,7 +103,7 @@ Flags:
       --mysqlctl_mycnf_template string                                   template file to use for generating the my.cnf file during server init
       --mysqlctl_socket string                                           socket file to use for remote mysqlctl actions (empty for local actions)
       --onclose_timeout duration                                         wait no more than this for OnClose handlers before stopping (default 10s)
-      --onterm_timeout duration                                          wait no more than this for OnTermSync handlers before stopping (default 10s)
+      --onterm_timeout duration                                          wait no more than this for OnTermSync handlers before stopping (default 5m10s)
```

Comment:
I assume this value is set to be 10 seconds greater than the shutdown timeout defined at
https://github.com/vitessio/vitess/blob/37e5c4ac9b1b93e1e363cb8470ecedc2b80c195d/go/vt/vttablet/tabletmanager/tm_init.go#L95

It may be better to define a global constant for 5 minutes and use it in both places to prevent drift in future.

Reply:
Not sure what you mean here. The additional 10 seconds was deliberate for some extra buffer time, so not the same for both. 

What would you suggest?

Reply:
something like:
```
var SharedTimeout = 5*time.Minute
...
mysqlShutdownTimeout = SharedTimeout
...
mysqlctldOnTermTimeout = SharedTimeout + 10*time.Second
```
It looks really ugly written out like this, but it establishes a correlation in code between these two timeouts instead of that being in our heads.
Of course now that I think about it, it won't prevent someone from changing mysqlShutdownTimeout to some other value anyway. So I'll leave it up to you whether we should really correlate these two timeouts in code in an explicit way.
A simpler way is to make `mysqlShutdownTimeout` exported, i.e. change it to `MysqlShutdownTimeout` and use it directly as the base for mysqlctld's on term timeout.
🤷

Reply:
I think the current version makes the dependency more clear that the timeout is how long we need for MySQL shutdown and a buffer? That’s imho less clear in that example and to me looks more accidental instead of deliberate.

Reply:
@deepthi Ah wait, I think I get the confusion. It's not really based on what `tm_init.go` has, but what is already in `mysqlctld`. But those could be shared, so it's less about this new change I think.

Reply:
@deepthi Did a little refactor in https://github.com/vitessio/vitess/pull/15575/commits/f1a508fc25ea76e219e7563fdd36aa25dc159d4e